CHANGE_LOG

```diff
@@ -1,3 +1,12 @@
+Version 0.57.0
```

Comment:
We don't normally add items to the change log as part of the PR. But this doesn't do any harm so I don't see a reason to necessarily purge it. @stuartarchibald @sklam what is your take?

Reply:
Some thoughts (unordered):
1. I think that capturing notes on important changes in the change log as PRs are merged is a good idea. 
2. There's no policy on incrementally adding things to the change log and I think it would probably make sense to develop one ahead of doing something ad-hoc which might set a undesired precedence.
3. The general approach of writing to the top of the change log would potentially lead to more merge conflicts and this adds a burden on to contributors/maintainers to resolve them.
4. At present the release notes/change log are written based on the PRs list scraped from the git log/what's in the milestone (should be the same!). It would be unlikely an alteration such as minimum version change be missed (especially as the version support matrix would need to reflect this and validating that is part of the release process).
5. In future, I think having a system that assimilates important release notes from specially named/formatted individual files (or similar) may be good approach and something worth discussion, but it is unlikely such a system would be ready for this PR even if accepted as an idea.

Given "automatic production of release notes" has also been informally discussed via other channels I'll raise this at the Numba maintainers' meeting next week. In the interests of getting this PR merged and making the change set minimal, I think simply removing it would be best for now.

Reply:
I added this to the changelog because this was something that you did @stuartarchibald in your commit that moved the minimum supported python version to 3.7 in https://github.com/numba/numba/commit/7bdd80023cc67a5896386ad8aa613ee26180b20f. I was trying to follow the best 😄 

I was also assuming it was because of 1) from above as it is a major change that people would be interested when developing numba along the `main` branch.

Reply:
> I added this to the changelog because this was something that you did @stuartarchibald in your commit that moved the minimum supported python version to 3.7 in [7bdd800](https://github.com/numba/numba/commit/7bdd80023cc67a5896386ad8aa613ee26180b20f). I was trying to follow the best smile. Seems like @sklam made a comment in agreement in [#6964 (comment)](https://github.com/numba/numba/pull/6964#discussion_r619423564).

Thanks for making the effort to form this patch based on an update of a prior similar change so as to make reviewing it easier, it's much appreciated! There indeed was an occurrence of an incremental write to the change log before, but I don't think it's happened since, therefore I would consider not doing it as having precedence. As noted above, this is of interest in relation to "automatic production of release notes" and will be discussed next week. If this something important to you then Numba also has weekly public meetings (details are pinned on Numba's discourse instance) where the schedule is open/community driven, you are welcome to raise it as a topic.

Reply:
I've pushed a commit that removes the changelog diff